hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java

```diff
@@ -131,6 +142,13 @@ public void initialize(URI name, Configuration conf) throws IOException {
         // If port number is not specified, read it from config
         omPort = OmUtils.getOmRpcPort(conf);
       }
+    } else if (OmUtils.isServiceIdsDefined(conf)) {
```

Comment:
Here conf passed might not be ozone configuration object, to get config values of ozone-site.xml, we need to convert conf to ozone configuration object.

`OmUtils.isServiceIdsDefined(new OzoneConfiguration(conf))`

Reply:
I wonder when would `conf` not be an `OzoneConfiguration`? I tested that in all my test cases, `conf` is always `OzoneConfiguration`.

Reply:
https://github.com/apache/hadoop/blob/trunk/hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java#L154 
This is code line calling creation of BasicOzoneClientAdapterImpl. And below is the code where it checks if conf passed is instance of OzoneConfiguration or not, if not convert to OzoneConfiguration object.

https://github.com/apache/hadoop/blob/trunk/hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneClientAdapterImpl.java#L112

Reply:
Make sense. Thanks! Fixing this in an upcoming commit.